pysam build fails for python 3.11 + macOS 12.6.7

(This is on a github action runner)
When installing dependencies in our testing environments, the build for pysam 0.21.0 fails for macOS 12.6.7 with python 3.11, but succeeds for python 3.10, as well as for python 3.11 with an Ubuntu 22.04.2 image.

Even though it finds `lzma.h`, configure reports that no liblzma development files are found:
```
  python -m pip install --upgrade pip
  python -m pip install poetry black flake8
  poetry install --no-interaction
  [...]
  # pysam: cython is available - using cythonize if necessary
  # pysam: htslib mode is shared
  # pysam: HTSLIB_CONFIGURE_OPTIONS=None
  # pysam: (sysconfig) CC=clang
  # pysam: (sysconfig) CFLAGS=-Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch arm64 -arch x86_64 -g 
  # pysam: (sysconfig) LDFLAGS=-arch arm64 -arch x86_64 -g
  [...]
  checking for lzma.h... yes
  checking for lzma_easy_buffer_encode in -llzma... no
  configure: error: liblzma development files not found
  
  The CRAM format may use LZMA2 compression, which is implemented in HTSlib
  by using compression routines from liblzma <http://tukaani.org/xz/>.
  
  Building HTSlib requires liblzma development files to be installed on the
  build machine; you may need to ensure a package such as liblzma-dev (on Debian
  or Ubuntu Linux), xz-devel (on RPM-based Linux distributions or Cygwin), or
  xz (via Homebrew on macOS) is installed; or build XZ Utils from source.
  
  Either configure with --disable-lzma (which will make some CRAM files
  produced elsewhere unreadable) or resolve this error to build HTSlib.
  [...]
  checking for lzma.h... yes
  checking for lzma_easy_buffer_encode in -llzma... no
  configure: error: liblzma development files not found
  
  The CRAM format may use LZMA2 compression, which is implemented in HTSlib
  by using compression routines from liblzma <http://tukaani.org/xz/>.
  
  Building HTSlib requires liblzma development files to be installed on the
  build machine; you may need to ensure a package such as liblzma-dev (on Debian
  or Ubuntu Linux), xz-devel (on RPM-based Linux distributions or Cygwin), or
  xz (via Homebrew on macOS) is installed; or build XZ Utils from source.
  
  Either configure with --disable-lzma (which will make some CRAM files
  produced elsewhere unreadable) or resolve this error to build HTSlib.
  Makefile:137: htscodecs.mk: No such file or directory
  config.mk:2: *** Resolve configure error first.  Stop.
  # pysam: htslib configure options: None
  ```

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -10,15 +10,13 @@ jobs:
         os: [ubuntu, macos]
         python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
         exclude:
-          # Run only the latest 3.x on macOS
+          # Run only the latest two 3.x versions on macOS
           - os: macos
             python-version: 3.7
           - os: macos
             python-version: 3.8
           - os: macos
             python-version: 3.9
-          - os: macos
-            python-version: 3.11
 
     steps:
       - name: Checkout pysam
@@ -32,11 +30,16 @@ jobs:
       - name: Install prerequisite Python libraries
         run:  pip install cython pytest pytest-pep8
 
-      - name: Install build prerequisites
+      - name: Install Linux build prerequisites
         if:   runner.os == 'Linux'
         run: |
           sudo apt-get update
           sudo apt-get install -q --no-install-recommends --no-install-suggests libcurl4-openssl-dev
+
+      - name: Update macOS build prerequisites
+        if:   runner.os == 'macOS'
+        run: |
+          brew unlink xz || true  # Remove brewed liblzma as it is not multiarch
 
       - name: Build (directly from checkout)
         run:  python setup.py build

